Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Endpoints/AWSPartitionDefinition.swift

```diff
@@ -7,232 +7,47 @@
-// build instead of being loaded from a static definition.
-public let awsPartitionJSON = """
-{
-  "version": "1.1",
-  "partitions": [
-    {
-      "id": "aws",
-      "regionRegex": "^(us|eu|ap|sa|ca|me|af|il)-\\\\w+-\\\\d+$",
-      "regions": {
-        "af-south-1": {
-          "description": "Africa (Cape Town)"
-        },
-        "ap-east-1": {
-          "description": "Asia Pacific (Hong Kong)"
-        },
-        "ap-northeast-1": {
-          "description": "Asia Pacific (Tokyo)"
-        },
-        "ap-northeast-2": {
-          "description": "Asia Pacific (Seoul)"
-        },
-        "ap-northeast-3": {
-          "description": "Asia Pacific (Osaka)"
-        },
-        "ap-south-1": {
-          "description": "Asia Pacific (Mumbai)"
-        },
-        "ap-south-2": {
-          "description": "Asia Pacific (Hyderabad)"
-        },
-        "ap-southeast-1": {
-          "description": "Asia Pacific (Singapore)"
-        },
-        "ap-southeast-2": {
-          "description": "Asia Pacific (Sydney)"
-        },
-        "ap-southeast-3": {
-          "description": "Asia Pacific (Jakarta)"
-        },
-        "ap-southeast-4": {
-          "description": "Asia Pacific (Melbourne)"
-        },
-        "aws-global": {
-          "description": "AWS Standard global region"
-        },
-        "ca-central-1": {
-          "description": "Canada (Central)"
-        },
-        "ca-west-1": {
-          "description": "Canada West (Calgary)"
-        },
-        "eu-central-1": {
-          "description": "Europe (Frankfurt)"
-        },
-        "eu-central-2": {
-          "description": "Europe (Zurich)"
-        },
-        "eu-north-1": {
-          "description": "Europe (Stockholm)"
-        },
-        "eu-south-1": {
-          "description": "Europe (Milan)"
-        },
-        "eu-south-2": {
-          "description": "Europe (Spain)"
-        },
-        "eu-west-1": {
-          "description": "Europe (Ireland)"
-        },
-        "eu-west-2": {
-          "description": "Europe (London)"
-        },
-        "eu-west-3": {
-          "description": "Europe (Paris)"
-        },
-        "il-central-1": {
-          "description": "Israel (Tel Aviv)"
-        },
-        "me-central-1": {
-          "description": "Middle East (UAE)"
-        },
-        "me-south-1": {
-          "description": "Middle East (Bahrain)"
-        },
-        "sa-east-1": {
-          "description": "South America (Sao Paulo)"
-        },
-        "us-east-1": {
-          "description": "US East (N. Virginia)"
-        },
-        "us-east-2": {
-          "description": "US East (Ohio)"
-        },
-        "us-west-1": {
-          "description": "US West (N. California)"
-        },
-        "us-west-2": {
-          "description": "US West (Oregon)"
+enum PartitionLoadingError: Error {
+    case fileNotFound
+    case readError(String)
+}
+
+public class PartitionData {
```

Comment:
I think if you just make this type an actor, it will enforce exclusive calls to `loadAWSPartitionJSON()`.

We should not be using DispatchQueue & other GCD types in new code, except where necessary to interact with pre-Swift concurrency code.

Reply:
I made PartitionDataCache an actor which is used by this and removed DispatchQueue